src/main/java/plugily/projects/buildbattle/handlers/themes/vote/VoteMenu.java

```diff
@@ -103,7 +104,14 @@ private NormalFastInv getGUI() {
 
     gui.addCloseHandler(event -> {
       if(arena.getArenaInGameStage() == BaseArena.ArenaInGameStage.THEME_VOTING) {
-        event.getPlayer().openInventory(event.getInventory());
+        plugin.getServer().getScheduler().runTaskLater(plugin, () -> {
+          HumanEntity humanEntity = event.getPlayer();
+          Inventory inventory = event.getInventory();
+
+          if (humanEntity.getOpenInventory().getTopInventory() != inventory) {
+            humanEntity.openInventory(inventory);
+          }
+        }, 3); // Delay operation to make sure the top inventory is closed before opening again
```

Comment:
does it work with one tick? (running on next tick)

Reply:
Seems to be working with `runTask`